app/code/Magento/CustomerGraphQl/Model/Resolver/CreateCustomer.php

```diff
@@ -0,0 +1,130 @@
+     * @var CustomerInterfaceFactory
+     */
+    private $customerFactory;
+
+    /**
+     * @var DataObjectHelper
+     */
+    private $dataObjectHelper;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+    /**
+     * @var SubscriberFactory
+     */
+    private $subscriberFactory;
+
+    /**
+     * @param DataObjectHelper $dataObjectHelper
+     * @param CustomerInterfaceFactory $customerFactory
+     * @param AccountManagementInterface $accountManagement
+     * @param StoreManagerInterface $storeManager
+     * @param SubscriberFactory $subscriberFactory
+     * @param CustomerDataProvider $customerDataProvider
+     */
+    public function __construct(
+        DataObjectHelper $dataObjectHelper,
+        CustomerInterfaceFactory $customerFactory,
+        AccountManagementInterface $accountManagement,
+        StoreManagerInterface $storeManager,
+        SubscriberFactory $subscriberFactory,
+        CustomerDataProvider $customerDataProvider
+    ) {
+        $this->customerDataProvider = $customerDataProvider;
+        $this->accountManagement = $accountManagement;
+        $this->customerFactory = $customerFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->storeManager = $storeManager;
+        $this->subscriberFactory = $subscriberFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($args['input']) || !is_array($args['input']) || empty($args['input'])) {
+            throw new GraphQlInputException(__('"input" value should be specified'));
+        }
+        try {
+            $customer = $this->createUserAccount($args);
+            $this->setUpUserContext($context, $customer);
+            if (array_key_exists('is_subscribed', $args['input'])) {
+                if ($args['input']['is_subscribed']) {
+                    $this->subscriberFactory->create()->subscribeCustomerById($customer->getId());
+                }
+            }
+            $data = $this->customerDataProvider->getCustomerById((int)$customer->getId());
+        } catch (LocalizedException $e) {
+            throw new GraphQlInputException(__($e->getMessage()));
+        }
+        return ['customer' => $data];
+    }
+
+    private function createUserAccount($args)
+    {
+        $customerDataObject = $this->customerFactory->create();
+        $this->dataObjectHelper->populateWithArray(
+            $customerDataObject,
+            $args['input'],
+            \Magento\Customer\Api\Data\CustomerInterface::class
+        );
+        $store = $this->storeManager->getStore();
+        $customerDataObject->setWebsiteId($store->getWebsiteId());
+        $customerDataObject->setStoreId($store->getId());
+
+        $password = array_key_exists('password', $args['input']) ? $args['input']['password'] : null;
```

Comment:
When do we not need a password?

Reply:
When we want to have registration and then send email to the customer with a link to set a password.
IN current state it works like that:
![image](https://user-images.githubusercontent.com/15912461/49404092-1ae17a00-f757-11e8-82ef-3e3f4ca1b8ce.png)